### PR TITLE
4dvarnet core ronan

### PR DESCRIPTION
**Pull request by rfablet**
Activation of dropout layer for each gradient-based update (dropout applied to gradient variables)
Add of a main file corresponding to SLA case-study without using torch lightning

### CODE REVIEW

**Comment by rfablet** [comment]
In this version, the code w/wo lightning leads to the same results in test mode (validated with onre pre-trained model). The training procedure of the lightning code reaches similar reconstruction performance in terms of monitored metrics (not yet validated on the test dataset).

**Comment by quentinf00** [comment]
L'idée est d'avoir 1 fichier que l'on mets à jours, je ne pense pas que ce soit une bonne idée de multiplier les fichiers comme ça. Peux tu intégrer les modifs de main_4DVarNN_SLA_withLightning.py au main.py et supprimer le main_4DVarNN_SLA_woLightning.py  ?

File: main_4DVarNN_SLA_withLightning.py [hunk 1]
```diff
@@ -0,0 +1,816 @@
...
+
+W = int(W/dx)
+
+# 
+UsePriodicBoundary = False # use a periodic boundary for all conv operators in the gradient model (see torch_4DVarNN_dinAE)
+InterpFlag         = False # True => force reconstructed field to observed data after each gradient-based update
+
+# Definiton of training, validation and test dataset
+# from dayly indices over a one-year time series
+iiTr1 = 0
```
Comment (MMAMAR) [inline main_4DVarNN_SLA_withLightning.py:61]: The indexes are defined two times.
